### ISSUE TITLE
Nested structs Causes assertion error

### ISSUE THREAD

**Issue by patmagee** [chanzuckerberg/miniwdl#127]
miniwdl version: `0.1.5`,`0.1.6`
also checked miniwdl: 0.1.4 and it seems to work fine.

@mlin I have been playing around with miniwdl alot and I think it is one of the best additions to the WDL community since cromwell! However I have come across this bug which makes the latest versions unusable:

A WDL which contains a `Struct` which contains another `Struct` that is in a "Complex" type, results in an `AssertionError` when loading the `WDLDocument`

A very simple example of this would be the following two structs:

```wdl
struct TestA {
    String mystring
    Array[TestB] myarray
}

struct TestB {
    Int num
}
```

When running `miniwdl check` or `WDL.load(...)` on a file containing these two structs, the following error is reported

```
Traceback (most recent call last):
  File "/home/patrick/development/dnastack/wdl-validator/venv/bin/miniwdl", line 11, in <module>
    sys.exit(main())
  File "/home/patrick/development/dnastack/wdl-validator/venv/lib/python3.6/site-packages/WDL/CLI.py", line 32, in main
    check(**vars(args))
  File "/home/patrick/development/dnastack/wdl-validator/venv/lib/python3.6/site-packages/WDL/CLI.py", line 96, in check
    doc = WDL.load(uri, path or [], check_quant=check_quant, import_uri=import_uri)
  File "/home/patrick/development/dnastack/wdl-validator/venv/lib/python3.6/site-packages/WDL/__init__.py", line 41, in load
    import_max_depth=import_max_depth,
  File "/home/patrick/development/dnastack/wdl-validator/venv/lib/python3.6/site-packages/WDL/Tree.py", line 1020, in load
    doc.typecheck(check_quant=check_quant)
  File "/home/patrick/development/dnastack/wdl-validator/venv/lib/python3.6/site-packages/WDL/Tree.py", line 950, in typecheck
    _initialize_struct_types(self.struct_types)
  File "/home/patrick/development/dnastack/wdl-validator/venv/lib/python3.6/site-packages/WDL/Tree.py", line 1456, in _initialize_struct_types
    b.rhs.type_id
  File "/home/patrick/development/dnastack/wdl-validator/venv/lib/python3.6/site-packages/WDL/Tree.py", line 76, in type_id
    return T._struct_type_id(self.members)
  File "/home/patrick/development/dnastack/wdl-validator/venv/lib/python3.6/site-packages/WDL/Type.py", line 399, in _struct_type_id
    ty = str(ty)
  File "/home/patrick/development/dnastack/wdl-validator/venv/lib/python3.6/site-packages/WDL/Type.py", line 205, in __str__
    + ("?" if self.optional else "")
  File "/home/patrick/development/dnastack/wdl-validator/venv/lib/python3.6/site-packages/WDL/Type.py", line 359, in __str__
    assert self.members
AssertionError
```

Please note, that this does work if instead of `TestA` aboce you have a struct that looks like the following:

```wdl
struct TestA {
  TestB myStruct
} 
```

**Comment by mlin** [comment]
Thanks! I'm traveling today but will look asap

**Comment by mlin** [comment]
@patmagee pls let us know if 52b99540bfba0df8c8aa425f06c4e3fe927498ab (included in v0.1.7) doesn't fix the problem on your end. Thanks a lot!
